eudemo/eudemo/equatorial_port.py

```diff
@@ -0,0 +1,229 @@
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+EU-DEMO Equatorial Port
+"""
+from dataclasses import dataclass
+from typing import Dict, List, Type, Union
+
+from numpy import array, ndarray
+
+from bluemira.base.builder import Builder, ComponentManager
+from bluemira.base.components import Component, PhysicalComponent
+from bluemira.base.designer import Designer
+from bluemira.base.parameter_frame import Parameter, ParameterFrame
+from bluemira.builders.tools import circular_pattern_component
+from bluemira.display.palettes import BLUE_PALETTE
+from bluemira.geometry.plane import BluemiraFace, BluemiraPlane
+from bluemira.geometry.tools import (
+    boolean_fuse,
+    extrude_shape,
+    make_polygon,
+    slice_shape,
+)
+from bluemira.geometry.wire import BluemiraWire
+
+
+class EquatorialPort(ComponentManager):
+    """
+    Wrapper around a Equatorial Port component tree
+    """
+
+    def xz_boundary(self) -> BluemiraWire:
+        """Returns a wire defining the x-z boundary of the Equatorial Port"""
+        return (
+            self.component.get_component("xz")
+            .get_component(EquatorialPortBuilder.KOZ)
+            .shape.boundary[0]
+        )
+
+
+@dataclass
+class EquatorialPortDesignerParams(ParameterFrame):
+    """
+    Equatorial Port Designer parameters
+    """
+
+    ep_height: Parameter[float]
+
+
+@dataclass
+class EquatorialPortBuilderParams(ParameterFrame):
+    """
+    Equatorial Port builder parameters
+    """
+
+    n_ep: Parameter[int]
+    ep_r_corner: Parameter[float]
+
+
+class EquatorialPortDesigner(Designer):
+    """
+    Equatorial Port Designer
+    """
+
+    param_cls: Type[EquatorialPortDesignerParams] = EquatorialPortDesignerParams
+
+    def __init__(
+        self,
+        params: Union[Dict, ParameterFrame, EquatorialPortDesignerParams],
+        build_config: Union[Dict, None],
+        x_ib: float,
+        x_ob: float,
+    ):
+        super().__init__(params, build_config)
+        self.x_ib = x_ib
+        self.x_ob = x_ob
+
+    def run(self) -> BluemiraWire:
+        """
+        Design the xz keep-out zone profile of the equatorial port
+        """
+        x_ib = self.x_ib
+        x_ob = self.x_ob
+        z_h = self.params.ep_height.value
+
+        x = (x_ib, x_ob, x_ob, x_ib)
+        z = (-z_h / 2, -z_h / 2, z_h / 2, z_h / 2)
+
+        ep_boundary = BluemiraFace(
+            make_polygon({"x": x, "y": 0, "z": z}, label="koz", closed=True)
+        )
+        return ep_boundary
+
+
+class EquatorialPortBuilder(Builder):
+    """
+    Equatorial Port Builder
+    """
+
+    NAME_XY = "Equatorial Ports xy"
```

Comment:
I dont think its necessary to have both a name_xy and name_xyz they are automatically put into the xy or xyz part of the tree:
```
EquatorialPort (Component)
├── xz (Component)
│   └── koz (PhysicalComponent)
├── xy (Component)
│   ├── Sector 1 (Component)
│   │   └── Equatorial Ports xy 1 (PhysicalComponent)
│   ├── Sector 2 (Component)
│   │   └── Equatorial Ports xy 2 (PhysicalComponent)
│   ├── Sector 3 (Component)
│   │   └── Equatorial Ports xy 3 (PhysicalComponent)
│   ├── Sector 4 (Component)
│   │   └── Equatorial Ports xy 4 (PhysicalComponent)
│   ├── Sector 5 (Component)
│   │   └── Equatorial Ports xy 5 (PhysicalComponent)
│   ├── Sector 6 (Component)
│   │   └── Equatorial Ports xy 6 (PhysicalComponent)
│   ├── Sector 7 (Component)
│   │   └── Equatorial Ports xy 7 (PhysicalComponent)
│   ├── Sector 8 (Component)
│   │   └── Equatorial Ports xy 8 (PhysicalComponent)
│   ├── Sector 9 (Component)
│   │   └── Equatorial Ports xy 9 (PhysicalComponent)
│   └── Sector 10 (Component)
│       └── Equatorial Ports xy 10 (PhysicalComponent)
└── xyz (Component)
    ├── Sector 1 (Component)
    │   └── Equatorial Ports xyz 1 (PhysicalComponent)

```

Reply:
Yes, but it means that both the xy profile components and the xyz 3d components get called `Equatorial Port 1` to ` Equatorial Port 10`. I've updated the tests so they're not retrieving the wrong component and asserting that the 2d xy profiles have no volume.